pkg/log/log.go

```diff
@@ -0,0 +1,29 @@
+package log
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+// Logger is a simple logging interface for Go.
+var Logger logr.Logger
+
+func init() {
+	// Build a zap development logger.
+	zapLog, err := zap.NewDevelopment()
+	if err != nil {
+		panic(fmt.Sprintf("error building logger: %v", err))
+	}
+	// zapr defines an implementation of the Logger
+	// interface built on top of Zap (go.uber.org/zap).
+	Logger = zapr.NewLogger(zapLog).WithName("cluster-ingress-operator")
+	Logger.Info("started zapr logger")
+
+	// Set a concrete logging implementation for all
+	// controller-runtime deferred Loggers.
+	log.SetLogger(Logger)
```

Comment:
Kind of a nit, but `init()` can be difficult to reason about. Alternative is to extract `SetLogger()` out to the entrypoint in `main.go`... as it stands, controller-runtime logging will only be wired if you happen to import our internal logging package, and if you wanted to reconfigure controller-runtime logging in a different context (e.g. tests) you would have to be careful about package loading order relative to your own `SetLogger` call which will compete with package init.